### PR TITLE
Stricter typings for event emitter and federated events

### LINKED ISSUE #7429: Better type classes that extends EventEmitter.
Hi, I think that you should better type the classes that extend EventEmitter, using Generics to list and strictly type the events available, instead of just having `string | symbol` for the name and `fn: (...args: Array<any>) => void` for the function.

There are a lot of libraries that provide classes with typed events and the one you are using, [`eventemitter3`](https://github.com/primus/eventemitter3) can also do it if you update the version you use.

----
### This is an example of what we have to do without modifying the PIXI to get the proper types :
```typescript
export type EventList = Record<string, any[]>;

export namespace EventEmitter {
	export type Emit<EventTypes extends EventList> = <K extends keyof EventTypes & string>(event: K, ...args: EventTypes[K]) => boolean;
	export type On<EventTypes extends EventList, This> = <K extends keyof EventTypes & string>(event: K, fn: (...params: EventTypes[K]) => void, context?: any) => This;
	export type Once<EventTypes extends EventList, This> = <K extends keyof EventTypes & string>(event: K, fn: (...params: EventTypes[K]) => void, context?: any) => This;
}

export interface Sprite {
	emit: EventEmitter.Emit<SpriteEvents>;
	on: EventEmitter.On<SpriteEvents, this>;
	once: EventEmitter.Once<SpriteEvents, this>;
}

export type SpriteEvents = {
	added: [container: PIXI.Container];
	click: [event: PIXI.InteractionEvent];
	mousedown: [event: PIXI.InteractionEvent];
	mousemove: [event: PIXI.InteractionEvent];
	mouseout: [event: PIXI.InteractionEvent];
	mouseover: [event: PIXI.InteractionEvent];
	mouseup: [event: PIXI.InteractionEvent];
	mouseupoutside: [event: PIXI.InteractionEvent];
	pointercancel: [event: PIXI.InteractionEvent];
	pointerdown: [event: PIXI.InteractionEvent];
	pointermove: [event: PIXI.InteractionEvent];
	pointerout: [event: PIXI.InteractionEvent];
	pointerover: [event: PIXI.InteractionEvent];
	pointertap: [event: PIXI.InteractionEvent];
	pointerup: [event: PIXI.InteractionEvent];
	pointerupoutside: [event: PIXI.InteractionEvent];
	removed: [container: PIXI.Container];
	removedFrom: [child: PIXI.DisplayObject, container: PIXI.Container, index: number];
	rightclick: [event: PIXI.InteractionEvent];
	rightdown: [event: PIXI.InteractionEvent];
	rightup: [event: PIXI.InteractionEvent];
	rightupoutside: [event: PIXI.InteractionEvent];
	tap: [event: PIXI.InteractionEvent];
	touchcancel: [event: PIXI.InteractionEvent];
	touchend: [event: PIXI.InteractionEvent];
	touchendoutside: [event: PIXI.InteractionEvent];
	touchmove: [event: PIXI.InteractionEvent];
	touchstart: [event: PIXI.InteractionEvent];
};
```

### PR DESCRIPTION
**Pull request by s-r-x**
##### Description of change

Closes #7429
##### Pre-Merge Checklist

- [ ] Tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: packages/accessibility/src/AccessibilityManager.ts [hunk 1]
```diff
@@ -222,7 +223,8 @@ export class AccessibilityManager
         self.document.removeEventListener('mousemove', this._onMouseMove, true);
         self.addEventListener('keydown', this._onKeyDown, false);
 
-        this.renderer.off('postrender', this.update);
+        // explicit cast to fix typescript 3.9.5 error - "this expression is not callable"
+        (this.renderer as AbstractRenderer).off('postrender', this.update);
```
Comment (ShukantPal) [inline packages/accessibility/src/AccessibilityManager.ts:227]: Why is this not callable without an explicit cast?

**Review by ShukantPal — CHANGES_REQUESTED** [review]

File: packages/accessibility/src/AccessibilityManager.ts (hunk 1 above)
Comment (s-r-x) [reply packages/accessibility/src/AccessibilityManager.ts:227]: no idea, in typescript 4.something there is no such error

**Review by ShukantPal — APPROVED** [review]

**Comment by bigtimebuddy** [comment]
@ShukantPal what about the events package? I think we need stricter typings there too for events.

**Comment by ShukantPal** [comment]
I think so, yes.

**Comment by littleboarx** [comment]
is there any possibility for the PR to be merged?

File: packages/accessibility/src/AccessibilityManager.ts (hunk 1 above)
Comment (SuperSodaSea) [reply packages/accessibility/src/AccessibilityManager.ts]: This is no longer an error after merging the latest dev branch.

**Review by SuperSodaSea — APPROVED** [review]
I merged the latest dev branch into it, removed the `@pixi/interaction` things & made a few fixes, and it still looks good. LGTM for v7!

**Pull request merged**